pkg/apis/agones/v1/gameserver.go

```diff
@@ -904,6 +904,7 @@ func (gs *GameServer) UpdateCounterCapacity(name string, capacity int64) error {
 	}
 	if counter, ok := gs.Status.Counters[name]; ok {
 		counter.Capacity = capacity
+		// TODO: If Capacity is now less than Count, do we want to reset Count here to equal Capacity?
```

Comment:
If Capacity is now less than Count, do we want to reset Count here (in UpdateCounterCapacity) to equal Capacity?

Reply:
This is a really good question. I think the answer should be "yes", since it's the most consistent experience. This might also be a "let's say 'yes' now, and see what the feedback is".

Reply:
One sticky point if we have the CounterAction:

Counter: {
  Count 11
  Capacity 20
}

CounterAction: {
  Decrement Count by 10
  Set Capacity to 10
}

Capacity gets set to 10, which then sets Count to also 10.
The Count then gets decremented by 10, which sets the count to 0.

(We could update Count first and then Capacity, but then we'd have issues on scale up when a customer wants to add Count and Capacity.)

Should we go ahead and do then, and just document that behavior?

Reply:
Yeah I think you are right - let's just pick an order and let's document it, and we can adjust later on if need be.